virtualenv.py

```diff
@@ -354,11 +356,11 @@ def copyfile(src, dest, symlink=True):
         logger.info('Copying to %s', dest)
         copyfileordir(src, dest, symlink)
 
-def writefile(dest, content, overwrite=True):
+def writefile(dest, content, overwrite=True, use_fsenc=False):
     if not os.path.exists(dest):
         logger.info('Writing %s', dest)
         with open(dest, 'wb') as f:
-            f.write(content.encode('utf-8'))
+            f.write(content.encode(sys.getfilesystemencoding() if use_fsenc else 'utf-8'))
```

Comment:
```sys.filesystemencoding()``` is not the correct encoding for bat files - you need to use sys.stdout.encoding. See the following (hopefully the Unicode characters will display correctly...):

```
C:\Work\Scratch>py                                                                                            
Python 3.5.2 (v3.5.2:4def2a2901a5, Jun 25 2016, 22:18:55) [MSC v.1900 64 bit (AMD64)] on win32                
Type "help", "copyright", "credits" or "license" for more information.                                        
>>> import sys                                                                                                
>>> sys.getfilesystemencoding()                                                                               
'mbcs'                                                                                                        
>>> sys.stdout.encoding                                                                                       
'cp437'                                                                                                       
>>> with open('x.bat', 'wb') as f:                                                                            
...   f.write("echo \N{POUND SIGN}\n".encode(sys.getfilesystemencoding()))                                    
...                                                                                                           
7                                                                                                             
>>>                                                                                                           
                                                                                                              
C:\Work\Scratch>x.bat                                                                                         
                                                                                                              
C:\Work\Scratch>echo ú                                                                                        
ú
```